medallion/exceptions.py

```diff
@@ -0,0 +1,46 @@
+# Medallion Custom Exceptions
+
+
+class MedallionError(Exception):
+    """base error class for Medallion
+
+    Args:
+        root_exception (Exception): Exception instance of root exception
+        desc (str): specific error description
+    """
+    def __init__(self, root_exception, desc=None):
+        self.root_exception = root_exception
+        self.desc = desc
+
```

Comment:
Consider defining `__str__` here as something like `"{0.desc}. Root exception: {0.root_exception}".format(self)`

Reply:
Works for me.